sdk/eventhub/azure-eventhubs/azure/eventhub/eventprocessor/checkpoint_manager.py

```diff
@@ -0,0 +1,30 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# -----------------------------------------------------------------------------------
+
+
+from .partition_manager import PartitionManager
+
+
+class CheckpointManager(object):
+    """Every PartitionProcessor has a CheckpointManager to save the partition's checkpoint.
+
+    """
+    def __init__(self, partition_id, eventhub_name, consumer_group_name, instance_id, partition_manager: PartitionManager):
+        self._partition_id = partition_id
+        self._eventhub_name = eventhub_name
+        self._consumer_group_name = consumer_group_name
+        self._instance_id = instance_id
+        self._partition_manager = partition_manager
+
+    async def update_checkpoint(self, offset, sequence_number):
```

Comment:
Are both offset and sequence_number required?

Reply:
@johanste both offset and sequence_number are enough to do checkpoint. Our design is to put them both in the checkpoint storage. What I heard was offset should be the one to do checkpoint whereas sequence_number is good information for debugging. Users will get confused whether to pass or not if we make "sequence_number" optional. And it's very easy to get offset and sequence_number. They both are EventData's properties.

@yunhaoling Python doesn't have method overload so I give up update_checkpoint(EventData). It would be more confusing to have update_checkpoint (EventData, offset, sequence_number).